src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -131,11 +138,11 @@ export class DefaultTelemetryService implements TelemetryService {
             const eventWithCommonMetadata = this.injectCommonMetadata(event, actualAwsContext)
             this._eventQueue.push(eventWithCommonMetadata)
             getLogger().debug(`telemetry: passive=${event.Passive ? 1 : 0} ${event.MetricName}`)
-        }
-    }
 
-    public get records(): ReadonlyArray<MetricDatum> {
-        return this._eventQueue
+            if (!isReleaseVersion()) {
```

Comment:
Can we tie this specifically to tests, or can we expose this functionality to release users? It's in a weird state where it's only for tests and it provides no additional benefit for development users (maybe we could drop a breakpoint here? but you don't need the logger for inspection), and could impact performance if lots of metadata is generated.

Reply:
It'd work fine in dev mode too, though are you thinking of some sort of UI? That could be interesting.

Reply:
It currently serves no purpose unless you're running a test against it, which isn't ideal if you're not testing. UI could be interesting (I think JB has something similar), but would probably be low priority since that kind of thing runs heavy in VS Code (easiest thing would just be a command palette command to the logs or something).

Reply:
Oh you mean dump the formatted metric to the log. I have that as a commit but haven't pushed it.